list.c

```diff
@@ -0,0 +1,186 @@
+#include <list.h>
+
+#define data 0
+#define next 1
+
+/*Prints list*/
+void StringListPrint(char** list) {
+
+    char** temp = list;
+    if (list == NULL) {
+        printf("List is empty!");
+        return;
+    }
+    printf("List members:\n");
+    while (temp != NULL) {
+        printf("%s\n", (temp[data]));
+        temp = temp[next];
+    }
+    printf("\n");
+}
+/* Destroy list and set pointer to NULL. */
+void StringListDestroy(char*** list) {
+
+    if (list == NULL) return;
+
+    char** prev = *list;
+    char** curr = (char**)prev[next];
+
+    /*iterate over list and delite curr prev element*/
+    while (curr != NULL) {
+        free(prev[data]);
+        free(prev);
+        prev = (char**)curr;
+        curr = (char**)curr[next];
+    }
+
+    free(prev[data]);
+    free(prev);
+
+    *list = NULL;
+}
+/* Inserts value at the end of the list. */
+void StringListAdd(char*** list, char* str) {
+
+    const size_t size = strlen(str) + 1;
+
+    //create new node
+    char** node = (char**)malloc(2 * sizeof(char*));
+    node[data] = (char*)malloc(sizeof(char) * size);
+    memcpy(node[data], str, sizeof(char) * size);
```

Comment:
The potential null pointer is passed into 'memcpy' function. Inspect the first argument.

Reply:
Resolved